### PR TITLE
Cognito user pools, identity pools and providers

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

**Review by michelvocks — APPROVED** [review]
LGTM

File: resources/cognito_identity_pools.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+				Type:        schema.TypeJSON,
+			},
+			{
+				Name:        "open_id_connect_provider_arn_s",
+				Description: "The ARNs of the OpenID Connect providers.",
+				Type:        schema.TypeStringArray,
+				Resolver:    schema.PathResolver("OpenIdConnectProviderARNs"),
+			},
+			{
+				Name:        "saml_provider_arn_s",
```
Comment (roneli) [inline resources/cognito_identity_pools.go:71]: saml_provider_arns

File: resources/cognito_identity_pools.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+				Description: "The \"domain\" by which Cognito will refer to your users.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "identity_pool_tags",
+				Description: "The tags that are assigned to the identity pool",
+				Type:        schema.TypeJSON,
+			},
+			{
+				Name:        "open_id_connect_provider_arn_s",
```
Comment (roneli) [inline resources/cognito_identity_pools.go:65]: open_id_connect_provider_arn_s -> open_id_connect_provider_arns

File: resources/cognito_user_pools.go [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+				Resolver:    schema.PathResolver("AdminCreateUserConfig.InviteMessageTemplate.EmailMessage"),
+			},
+			{
+				Name:        "admin_create_user_config_invite_message_template_email_subject",
+				Description: "The subject line for email messages",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("AdminCreateUserConfig.InviteMessageTemplate.EmailSubject"),
+			},
+			{
+				Name:        "admin_create_user_config_invite_message_template_sms_message",
```
Comment (roneli) [inline resources/cognito_user_pools.go:61]: would remove message suffix

Comment (roneli) [inline resources/cognito_user_pools.go:61]: I would rename these columns to a short name based on the meaning they are too long.

File: resources/cognito_user_pools.go [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "device_configuration_challenge_required_on_new_device",
+				Description: "Indicates whether a challenge is required on a new device",
+				Type:        schema.TypeBool,
+				Resolver:    schema.PathResolver("DeviceConfiguration.ChallengeRequiredOnNewDevice"),
+			},
+			{
+				Name:        "device_configuration_device_only_remembered_on_user_prompt",
```
Comment (roneli) [inline resources/cognito_user_pools.go:104]: would remove device_configuration prefix

File: resources/cognito_user_pools.go [hunk 5]
```diff
@@ -0,0 +1,612 @@
...
+				Type:        schema.TypeBool,
+				Resolver:    schema.PathResolver("DeviceConfiguration.DeviceOnlyRememberedOnUserPrompt"),
+			},
+			{
+				Name:        "domain",
+				Description: "Holds the domain prefix if the user pool has a domain associated with it.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "email_configuration_configuration_set",
```
Comment (roneli) [inline resources/cognito_user_pools.go:115]: configuration_configuration redundant

File: resources/cognito_user_pools.go [hunk 6]
```diff
@@ -0,0 +1,612 @@
...
+				Description: "The set of configuration rules that can be applied to emails sent using Amazon SES",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("EmailConfiguration.ConfigurationSet"),
+			},
+			{
+				Name:        "email_configuration_email_sending_account",
```
Comment (roneli) [inline resources/cognito_user_pools.go:121]: email_configuration_sending_account

File: resources/cognito_user_pools.go [hunk 7]
```diff
@@ -0,0 +1,612 @@
...
+				Resolver:    schema.PathResolver("EmailConfiguration.EmailSendingAccount"),
+			},
+			{
+				Name:        "email_configuration_from",
+				Description: "Identifies either the sender’s email address or the sender’s name with their email address",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("EmailConfiguration.From"),
+			},
+			{
+				Name:        "email_configuration_reply_to_email_address",
```
Comment (roneli) [inline resources/cognito_user_pools.go:133]: reply_to_address

**Review by roneli — CHANGES_REQUESTED** [review]
Looks Good, I would change a few column names that feel too long or have redundant prefixes/suffixes

File: resources/cognito_identity_pools.go (hunk 1 above)
Comment (irmatov) [reply resources/cognito_identity_pools.go:71]: Done

File: resources/cognito_identity_pools.go (hunk 2 above)
Comment (irmatov) [reply resources/cognito_identity_pools.go:65]: Done

File: resources/cognito_user_pools.go (hunk 3 above)
Comment (irmatov) [reply resources/cognito_user_pools.go:61]: Done

Comment (irmatov) [reply resources/cognito_user_pools.go:61]: Shortened field names as suggested.

File: resources/cognito_user_pools.go (hunk 4 above)
Comment (irmatov) [reply resources/cognito_user_pools.go:104]: Done

File: resources/cognito_user_pools.go (hunk 6 above)
Comment (irmatov) [reply resources/cognito_user_pools.go:121]: Done

File: resources/cognito_user_pools.go (hunk 7 above)
Comment (irmatov) [reply resources/cognito_user_pools.go:133]: done

File: resources/cognito_user_pools.go (hunk 5 above)
Comment (irmatov) [reply resources/cognito_user_pools.go:115]: Done

**Review by roneli — APPROVED** [review]
Looks Great!

**Pull request merged**